### PR TITLE
chore: cleanup minor core data things [WPB-5993]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5993" title="WPB-5993" target="_blank">BugWPB-5993</a>  [iOS] fix core data migrating corrupted ParticipantRole objects
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Just some minor ugly things to cleanup, that I've found during my long core data analysis.
- Add a unit test for `nil` identifier.

#### Test Coverage

- [x] I have added automated test to this contribution

#### How to Test

- Nothing more to test

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-system/Source/ZMSLog.swift [hunk 1]
```diff
@@ -376,7 +376,7 @@ extension ZMSLog {
 
         do {
             let data = Data(string.utf8)
-            try updatingHandle?.wr_write(data)
+            try updatingHandle?.write(contentsOf: data)
```
Comment (markusfassbender) [inline wire-ios-system/Source/ZMSLog.swift:379]: See deprecation from iOS 13 and later: https://developer.apple.com/documentation/foundation/nsfilehandle/1410936-writedata?language=objc

**Review by caldrian — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]
